dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleAPIImpl.java

```diff
@@ -82,7 +111,234 @@ public Bundle getBundleById(String id) throws DotDataException {
+
+		Logger.info(this, "Removing bundle: " + bundleId + " and dependencies by: " + user.getUserId());
+
+		try {
+
+			Logger.info(this, "Removing audit status for a bundle: " + bundleId);
+			this.publishAuditAPI.deletePublishAuditStatus(bundleId);
+		} catch (DotPublisherException e) {
+
+			throw new DotDataException(e);
+		}
+
+		Logger.info(this, "Removing all pushed assets for a bundle: " + bundleId);
+		this.pushedAssetsFactory.deletePushedAssetsByBundle(bundleId);
+
+		Logger.info(this, "Removing all assets from bundle: " + bundleId);
+		this.bundleFactory.deleteAllAssetsFromBundle(bundleId);
+
+		Logger.info(this, "Removing environments from bundle: " + bundleId);
+		this.bundleFactory.deleteBundleEnvironmentByBundle(bundleId);
+
+		Logger.info(this, "Removing bundle: " + bundleId);
+		this.deleteBundle(bundleId);
+	}
+
+	@WrapInTransaction
+	@Override
+	public void deleteBundleAndDependencies(final String bundleId, final User user) throws DotDataException {
+
+		final Bundle bundle = this.getBundleById(bundleId);
+		if (null != bundle) {
+
+			this.deleteBundleAndDependencies(bundle, user);
+		} else {
+
+			throw new NotFoundInDbException("The bundle id: " + bundleId + " does not exists");
+		}
+	}
+
+	private void validateBundleDeletePermission(final User user, final Bundle bundle) throws DotDataException {
+
+		/**
+		if (!APILocator.getPermissionAPI().doesUserHavePermission(bundle,
+				PermissionAPI.PERMISSION_EDIT, user, false)) {
+
+			throw new DotSecurityException("User : " + user.getUserId() +
+					", is not allowed to delete the bundle: " + bundleId)
+		}*/
+		// todo: bundle is not a permissionable yet, so can not validate yet.
+	}
+
+	@WrapInTransaction
+	@Override
+	public Set<String> deleteBundleAndDependenciesOlderThan(final Date olderThan, final User user) throws DotDataException {
+
+		final ImmutableSet.Builder<String> bundlesDeleted = new ImmutableSet.Builder<>();
+		final boolean isAdmin = this.userAPI.isCMSAdmin(user);
+
+		try {
+
+			Logger.info(this, "Deleting bundles older than: " + olderThan);
+			try (final Stream<String> bundleIds = this.getOlderBundleIds(olderThan, user.getUserId(), isAdmin)) {
+
+				bundleIds.forEachOrdered( bundleId -> {
+
+					this.internalDeleteBundleAndDependencies(bundleId, user);
+					bundlesDeleted.add(bundleId);
+				});
+			}
+		} catch (IOException e) {
+
+			throw new DotDataException(e);
+		}
+
+		return bundlesDeleted.build();
+	}
+
+	private Stream<String> getOlderBundleIds(final Date olderThan, final String userId, final boolean isAdmin)
+			throws IOException, DotDataException {
+
+		final File tempFile       = FileUtil.createTemporalFile("bundle-older-delete");
+		final int limit           = 100;
+		int offset                = 0;
+		List<Bundle> sentBundles  = isAdmin?
+				this.bundleFactory.findSentBundles(olderThan, limit, offset):
+				this.bundleFactory.findSentBundles(olderThan, userId, limit, offset);
+		BufferedWriter fileWriter = null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ensure that resources like this BufferedWriter object are closed after use](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4502981)